passes/hierarchy/hierarchy.cc

```diff
@@ -1093,18 +1134,20 @@ struct HierarchyPass : public Pass {
 				log("Mapping positional arguments of cell %s.%s (%s).\n",
 						RTLIL::id2cstr(module->name), RTLIL::id2cstr(cell->name), RTLIL::id2cstr(cell->type));
 				dict<RTLIL::IdString, RTLIL::SigSpec> new_connections;
-				for (auto &conn : cell->connections())
-					if (conn.first[0] == '$' && '0' <= conn.first[1] && conn.first[1] <= '9') {
-						int id = atoi(conn.first.c_str()+1);
+				for (auto &conn : cell->connections()) {
+					int id;
+					if (read_id_num(conn.first, &id)) {
 						std::pair<RTLIL::Module*,int> key(design->module(cell->type), id);
 						if (pos_map.count(key) == 0) {
 							log("  Failed to map positional argument %d of cell %s.%s (%s).\n",
-									id, RTLIL::id2cstr(module->name), RTLIL::id2cstr(cell->name), RTLIL::id2cstr(cell->type));
+							    id, RTLIL::id2cstr(module->name), RTLIL::id2cstr(cell->name), RTLIL::id2cstr(cell->type));
 							new_connections[conn.first] = conn.second;
 						} else
 							new_connections[pos_map.at(key)] = conn.second;
-					} else
+					} else {
```

Comment:
It seems like some of these changes are not related to the actual `atoi` => `read_id_num` improvement. I'd prefer we minimize the indentation/bracing changes around here to just those necessary. Otherwise, LGTM.

Reply:
I can drop them if you'd like, but had to add some braces above to avoid things getting mismatched (there was `for (A) if (B) { C } else D`, which I stupidly changed to `for (A) B0; if (B1) { C } else D` which... doesn't work!). Once I'd added the necessary braces, it looked weirdly wonky not to add them for the `else` side. What do you think?

Reply:
My initial thought was to just do the minimal change, despite the current un-braced `else` branch looking a little funny. You could also consider moving up the declaration of `id` to avoid changing any of the irrelevant branches.

Reply:
Ok, I'll drop the later braces. Let's keep `id` with the minimal scope: no need to pollute a larger namespace than necessary.